### ISSUE TITLE
Play count keeps resetting

### ISSUE THREAD

**Issue by onebrownfinch** [davidjb/itunes-change-play-count#3]
Hello -- thank you so much for writing this script, it's super easy to use! I'm not sure if you'd be able to help, but basically I've run into a problem where the play count is reset to its former value after some time (usually only a minute or two). I suspect it has something to do with the iCloud Music Library, as I'm an Apple Music subscriber, so idk if you can help here. It doesn't matter whether the manually input play count is higher or lower than the preexisting one, nor does it matter whether the track is uploaded or matched / downloaded from AM. I've tried using your script and then playing and skipping to the end of the track, but this results in the same problem plus one extra play. Also, initiating a sync of my cloud library just makes the problem pop up even more quickly, and I've tried quitting out of AM on my phone prior to changing the play count, as well as using Apple's "reset play count to 0" feature (which, itself, works) prior to running the script, both to no avail.

Any insight? Thanks in advance!

**Comment by davidjb** [comment]
You're very welcome - thanks for the compliment.

It sounds like your issue is the same or similar to one that was described previously - https://github.com/davidjb/itunes-change-play-count/issues/1#issuecomment-1094409507. I don't use a cloud music library so haven't experienced this myself or know of a workaround, sorry. If you come across a solution, then it could potentially be factored into the AppleScript. PRs are most definitely welcome 👍

**Comment by onebrownfinch** [comment]
Yeah, I did some digging after adding this comment and realized that it indeed is an AM / cloud library issue. I'm an AppleScript newbie so I'll likely be tinkering with it for some time before I come up with any useful solution, but I have a good idea on how to maybe subvert the issue. Thanks for your prompt reply!

**Comment by onebrownfinch** [comment]
If you're at all interested, I spent some time writing a bit of code that seems to work okay...it's built around the idea that skipping to the tail end of a track and playing it will actually update the play count. As I said, I'm no AppleScript professional, just someone who codes like 4 times a year for fun, so I'm sure it's a bit janky and there's a better way to do parts of it. If you get the chance to look at it / have some thoughts on how to make it better, I'm all ears!

Some bugs that I've noticed pop up sometimes:
- Code adds one too few or one too many plays
- Code stops after a few runs for no discernible reason
- Code never runs in the first place -- this one I'm pretty sure is a Music issue, rather than an AppleScript one. Music doesn't like repeatedly playing and fast-forwarding tracks. Restarting Music usually fixes this.

Edit: I updated the code to work with your "apply to all" feature.
**Edit: Okay, I think I've gotten the code to work without bugging out on me. I've run it multiple times without incurring any of the issues I mentioned, but unfortunately it still is a bit of a slow-mover, at least on my computer.**

[Play Count Sandbox 3.0.zip](https://github.com/davidjb/itunes-change-play-count/files/10143440/Play.Count.Sandbox.3.0.zip)

**Comment by dburlington** [comment]
Only 3 1/2 years later but I finally discovered this. Works like a charm with Apple Cloud and Match. Took about 4 seconds to run on a double CD needed to restore to 19 plays. Nicely done!
